ssl/quic/quic_srt_gen.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+#include "internal/quic_srt_gen.h"
+#include <openssl/core_names.h>
+#include <openssl/evp.h>
+
+struct quic_srt_gen_st {
+    EVP_MAC         *mac;
+    EVP_MAC_CTX     *mac_ctx;
+};
+
+/*
+ * Simple HMAC-SHA256-based stateless reset token generator.
```

Comment:
Bikeshed: Should we use something more modern such as KMAC with SHAKE-128?

Reply:
I suggested AES ECB with random key ...

Reply:
KMAC's pretty slow currently and less well tested.

GMAC might be worth consideration: it's just raw AES GCM.

Reply:
GMAC should not be IMO used in this kind of contexts.